### PR TITLE
Validate shell scripts

### PR DESCRIPTION
**Pull request by skitt**
Since we’re focusing on shell scripts rather than Go code for our
infrastructure, it seems useful to add some validation of shell
scripts using ShellCheck.

This adds scripts/shared/lib/* for now.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: scripts/shared/lib/utils [hunk 1]
```diff
@@ -34,15 +38,16 @@ function with_context() {
 # 2nd argument is the command to execute, which will have the $cluster variable set.
 # 3rd argument and so forth get passed to the command.
 function run_parallel() {
-    local clusters=$(eval echo "$1")
-    local cmnd=$2
+    local clusters cmnd
+    cluster=$(eval echo "$1")
```
Comment (mkolesnik) [inline scripts/shared/lib/utils]: Should be clusters

Comment (skitt) [reply scripts/shared/lib/utils]: Thanks, fixed!

**Comment by skitt** [comment]
> However, maybe I'm missing something but I don't see that you added execution of the tool?

See `scripts/validate`, it’s run there (and fails the build if an error creeps in).

File: scripts/shared/lib/utils [hunk 2]
```diff
@@ -70,5 +76,6 @@ function declare_cidrs() {
 }
 
 function declare_kubeconfig() {
-    export KUBECONFIG=$(echo ${KUBECONFIGS_DIR}/kind-config-cluster{1..3} | sed 's/ /:/g')
+    KUBECONFIG=$(echo "${KUBECONFIGS_DIR}"/kind-config-cluster{1..3} | sed 's/ /:/g')
+    export KUBECONFIG
```
[the remark this answers is not in the document]
Comment (skitt) [reply scripts/shared/lib/utils]: Done, thanks!

**Pull request merged**